tests/integration/test_charm.py

```diff
@@ -68,3 +75,85 @@ async def test_owasp_modsecurity_custom_rules(app_url_modsec_ignore: str):
     """
     response = requests.get(app_url_modsec_ignore)
     assert response.status_code == 200
+
+
+async def test_owasp_modsecurity_crs_relation(ops_test: OpsTest, app_name: str, tmp_path: Path):
+    """
+    arrange: given charm has been built and deployed
+    act: relate the ingress integrator with any-charm and use any-charm to toggle modsecurity
+        option through relation via ingress lib.
+    assert: modsecurity annotations should be attached and detached from the kubernetes ingress
+        resource according to the modsecurity option.
+    """
+    await ops_test.model.applications[app_name].set_config(
+        {"owasp-modsecurity-crs": "false", "owasp-modsecurity-custom-rules": ""}
+    )
+    any_charm_tmp_path = tmp_path / "any-charm"
+    if not any_charm_tmp_path.exists():
+        subprocess.run(  # nosec
+            ["git", "clone", "https://github.com/canonical/any-charm.git", any_charm_tmp_path]
+        )
+
+    any_charm = await ops_test.build_charm(any_charm_tmp_path)
+    ingress_lib = Path("lib/charms/nginx_ingress_integrator/v0/ingress.py").read_text()
+    any_charm_src_overwrite = {
+        "ingress.py": ingress_lib,
+        "any_charm.py": dedent(
+            """\
+        from ingress import IngressRequires
+        from any_charm_base import AnyCharmBase
+        class AnyCharm(AnyCharmBase):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                self.ingress = IngressRequires(
+                    self,
+                    {
+                        "service-hostname": "any",
+                        "service-name": self.app.name,
+                        "service-port": 80,
+                        "owasp-modsecurity-crs": True
+                    }
+                )
+            def update_ingress(self, ingress_config):
+                self.ingress.update_config(ingress_config)
+        """
+        ),
+    }
+    await ops_test.model.deploy(
+        str(any_charm),
+        application_name="any",
+        series="focal",
+        config={"src-overwrite": json.dumps(any_charm_src_overwrite)},
+    )
+    await ops_test.model.add_relation("any", f"{app_name}:ingress")
+    await ops_test.model.wait_for_idle(status=ActiveStatus.name)
+
+    kubernetes.config.load_kube_config()
+    kube = kubernetes.client.NetworkingV1Api()
+    ingress_annotations = kube.read_namespaced_ingress(
+        "any-ingress", namespace=ops_test.model.name
+    ).metadata.annotations
+    assert ingress_annotations["nginx.ingress.kubernetes.io/enable-modsecurity"] == "true"
+    assert (
```

Comment:
Why is this assert in brackets whereas the other asserts aren't?

Reply:
black use this trick to shorten the lines.